### PR TITLE
prometheus: drop time-bucket series, increase ram limit to 2G

### PR DESCRIPTION
**Pull request by eduser25**
(tentative)

We are using large amounts of labels to tag our metrics with addtional
dimensions of interest. This is good for visibility, however this explodes
in RAM consumption on Prometheus once scraped.

Most of the heavily unscaling metrics are the time-bucket series, which
already have large number of dimensions due to the buckets themselves.

- By filtering out time-bucket series (which might contain latency and
request total time buckets), we can cut the memory consumption around
66~70%, allowing larger scales to work under 2Gb, compared with same
scale crashing at 4Gb).
- Additionally, rose the RAM limit for prometheus to 2Gb.

Purely tentative, proper discussion on relabeling has to happen, this
is however required to even support slightly larger number of pods for
the time being.

More info:
#1934 

Signed-off-by: Eduard Serra <eduser25@gmail.com>

- Metrics                [X]
- Performance            [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: charts/osm/templates/prometheus-configmap.yaml [hunk 1]
```diff
@@ -47,7 +51,11 @@ data:
       - job_name: 'kubernetes-pods'
         kubernetes_sd_configs:
         - role: pod
-        relabel_configs:
+        metric_relabel_configs:
+        - source_labels: [__name__]
+          regex: .*bucket.*
```
Comment (nojnhuh) [inline charts/osm/templates/prometheus-configmap.yaml:56]: We will need `osm_request_duration_ms_bucket` for SMI metrics eventually. Is there an easy way to ignore all but that one?

Comment (eduser25) [reply charts/osm/templates/prometheus-configmap.yaml:56]: I think the proper solution might be a ~whitelist~ allowed list for what we explicitly want. Didn't go the extra mile as I'm onto something else, but if we feel we don't want this at all, I will not push for this either.

Comment (nojnhuh) [reply charts/osm/templates/prometheus-configmap.yaml:56]: SMI metrics is far enough out that it shouldn't bock this change. Once we get to that point, it should be simple enough to get it to work.

**Review by nojnhuh — APPROVED** [review]

**Review by michelleN — APPROVED** [review]
Does this change have any implications on the grafana dashboards? would the screenshots in docs change at all?

**Comment by eduser25** [comment]
> Does this change have any implications on the grafana dashboards? would the screenshots in docs change at all?

Yes it does. It kills the request and response latency stats as those are buckets. I wonder if we want this commit as is right now.
The problem is, the current labeling scales so incredibly poorly that I can't even use prometheus as it just runs out of memory even at 4Gb mem.

I can just cherrypick for my own testing, but I wonder what some other user might think if they happen to experience the same. Should we care for the time being? or should we open and mark this as a must-fix bug sometime soon? @michelleN

**Comment by eduser25** [comment]
https://github.com/openservicemesh/osm/issues/2122

**Pull request merged**